cpp/src/orderby/is_sorted.cu

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/cudf.h>
+#include <utilities/cudf_utils.h>
+#include <utilities/error_utils.hpp>
+
+#include <table/device_table.cuh>
+#include <table/device_table_row_operators.cuh>
+
+#include <rmm/thrust_rmm_allocator.h>
+
+/* ----------------------------------------------------------------------------*/
+/**
+ * @brief Wrapper around row_inequality_comparator to pass row and row+1
+ */
+/* ----------------------------------------------------------------------------*/
+
+template <bool nullable = true>
```

Comment:
Put everything in `namespace cudf`, and put the comparator struct in `namespace cudf::detail`.

Reply:
As is_sorted is being used, no need of comparator, so it will be removed. But if you think it can be used somewhere else, I will keep it in in predicates.hpp .